src/config/config_impl.go

```diff
@@ -252,6 +267,28 @@ func (this *rateLimitConfigImpl) GetLimit(
 		return rateLimit
 	}
 
+	if descriptor.GetLimit() != nil {
+		rateLimitKey := domain + "." + this.descriptorToKey(descriptor)
+		rateLimit := value.overrides[rateLimitKey]
+		rateLimitOverrideUnit := pb.RateLimitResponse_RateLimit_Unit(descriptor.GetLimit().GetUnit())
+		if rateLimit != nil {
+			if rateLimit.Limit.RequestsPerUnit != descriptor.GetLimit().GetRequestsPerUnit() ||
+				rateLimit.Limit.Unit != rateLimitOverrideUnit {
+
+				rateLimit.Limit.RequestsPerUnit = descriptor.GetLimit().GetRequestsPerUnit()
+				rateLimit.Limit.Unit = rateLimitOverrideUnit
```

Comment:
Aren't there potential concurrency issues here and below? I'm wondering if this is premature optimization and we should just allocate a new limit and return it to avoid potential issues?

Reply:
Ok, makes sense, thank you for the review. Removed the caching for now, in case we discover perf issues with allocating new structs every time I'll come back with a followup.